includes/utils/EnvironmentChecker.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * EnvironmentChecker.php
+ *
+ * Checks to see if the environment matches the passed conditions.
+ * Supported conditions include:
+ *
+ * - EDD version number -- either specific versions or wildcards (e.g. "2.x").
+ * - Type of license (pass level, à la carte, free).
+ *
+ * @package   easy-digital-downloads
+ * @copyright Copyright (c) 2021, Easy Digital Downloads
+ * @license   GPL2+
+ * @since     2.11.4
+ */
+
+namespace EDD\Utils;
+
+use EDD\Admin\Pass_Manager;
+
+class EnvironmentChecker {
+
+	/**
+	 * @var Pass_Manager
+	 */
+	protected $passManager;
+
+	/**
+	 * Number of EDD license keys entered.
+	 *
+	 * @since 2.11.4
+	 *
+	 * @var int
+	 */
+	protected $numberLicenseKeys;
+
+	/**
+	 * Types of license/pass conditions that we support.
+	 * The key is the condition slug and the value is the corresponding
+	 * method to call in the `Pass_Manager` class to check the condition.
+	 *
+	 * @since 2.11.4
+	 *
+	 * @var string[]
+	 */
+	protected $validLicenseConditions = array(
+		'free'              => 'isFree',
+		'ala-carte'         => 'hasIndividualLicense',
+		'pass-personal'     => 'hasPersonalPass',
+		'pass-extended'     => 'hasExtendedPass',
+		'pass-professional' => 'hasProfessionalPass',
+		'pass-all-access'   => 'hasAllAccessPass',
+		'pass-any'          => 'has_pass',
+	);
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->passManager = new Pass_Manager();
+
+		global $edd_licensed_products;
+
+		$this->numberLicenseKeys = is_array( $edd_licensed_products ) ? count( $edd_licensed_products ) : 0;
```

Comment:
Should this use the new ExtensionRegistry instead? Or does that not work because of extensions not using it yet?

Reply:
Actually I don't think I need this here at all anymore. It was before we added the new helper methods to the pass manager I think... Not actually seeing `$numberLicenseKeys` used anywhere anymore. I'm going to remove it. Hopefully nothing breaks.

Reply:
It took me a while to clean out license keys and get my environment to be `isFree` so I could get the test notification but I did. Removing this doesn't seem to have broken anything!